modules/relay/src/main/RelayApi.scala

```diff
@@ -138,7 +138,20 @@ final class RelayApi(
     tourRepo.coll.insert.one(tour) inject tour
 
   def tourUpdate(tour: RelayTour, data: RelayTourForm.Data)(using Me): Funit =
-    tourRepo.coll.update.one($id(tour.id), data.update(tour)).void andDo
+    tourRepo.coll.update
+      .one(
+        $id(tour.id),
+        $set(
+          "name"            -> data.name,
+          "description"     -> data.description,
+          "markup"          -> data.markup,
+          "tier"            -> data.tier,
+          "autoLeaderboard" -> data.autoLeaderboard,
+          "players"         -> data.players,
+          "spotlight"       -> data.spotlight
+        )
+      )
```

Comment:
this loses the logic performed by `data.update` and introduces more bugs

```scala
    def update(tour: RelayTour)(using Me) =
      tour
        .copy(
          name = name,
          description = description,
          markup = markup,
          tier = tier ifTrue Granter(_.Relay),
          autoLeaderboard = autoLeaderboard,
          players = players,
          spotlight = spotlight.filterNot(_.isEmpty)
        )
        .reAssignIfOfficial
```

Reply:
maybe something like this would work and also avoid repeating the field names:

```scala
tourRepo.coll.update.one($id(tour.id), $set(data.update(tour)))
```

Reply:
i'm pretty sure what i did is bad, and i am watching golf channel for tips